src/nlsolve/newton.jl

```diff
@@ -54,6 +54,7 @@ Equations II, Springer Series in Computational Mathematics. ISBN
   else
     dz = _reshape(W \ _vec(ztmp), axes(ztmp))
   end
+  integrator.destats.nsolve += 1
```

Comment:
This is currently incrementing the linear solve counter on non-linear solves. I guess we should have two counters - each for linear and nonlinear.

Reply:
Oh my bad, still this doesn't counts the linear solves in the Rosenbrocks.

Reply:
yeah, we can keep improving this